File: bme_ros_tutorials/scripts/publisher_custom.py
#!/usr/bin/env python3

import rospy
from bme_ros_tutorials.msg import Test # Our custom message type used in the node

rospy.init_node('publisher_custom')    # Init the node with name "publisher_custom"

pub = rospy.Publisher('publisher_custom_topic', Test, queue_size=10)

rospy.loginfo("Custom publisher Python node has started and publishing data on publisher_custom_topic")

rate = rospy.Rate(5) # 5Hz

testMessage = Test()
testMessage.first_name = "David"
testMessage.last_name  = "Dudas"
testMessage.age        = 34
testMessage.score      = 0

while not rospy.is_shutdown(): # Run the node until Ctrl-C is pressed

    pub.publish(testMessage)   # Publishing data on topic "publisher_custom_topic"
    
    testMessage.score += 1
    
    rate.sleep()               # The loop runs at 5Hz

File: bme_ros_tutorials/src/publisher_custom.cpp
#include "bme_ros_tutorials/Test.h" // Our custom message type used in the node
#include "ros/ros.h"

int main(int argc, char **argv) {
  // Init the node with name "publisher_custom"
  ros::init(argc, argv, "publisher_custom");
  ros::NodeHandle nh; // NodeHandle will fully initialze the node

  // Advertise the ROS topic
  ros::Publisher pub =
      nh.advertise<bme_ros_tutorials::Test>("publisher_custom_topic", 10);

  ROS_INFO("Custom publisher C++ node has started and publishing data on "
           "publisher_custom_topic");

  ros::Rate loop_rate(5); // 5 Hz

  bme_ros_tutorials::Test test_message;
  // test_message is now our custom ROS Test type variable
  // that is ready to be published

  // assign default values to the message fields
  test_message.first_name = "David"; // Initializing first name string
  test_message.last_name = "Dudas";  // Initializing last name string
  test_message.age = 32;             // Initializing age
  test_message.score = 0; // Initializing score that we'll increment later

  while (ros::ok()) // Run the node until Ctrl-C is pressed
  {
    // Publishing data on topic "publisher_custom_topic"
    pub.publish(test_message);

    test_message.score++;
    loop_rate.sleep(); // The loop runs at 5Hz
  }
}
